src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -375,20 +381,30 @@ private void handleOnDragDropped(TableRow<BibEntryTableViewModel> row, BibEntryT
             switch (ControlHelper.getDroppingMouseLocation(row, event)) {
                 case TOP, BOTTOM -> importHandler.importFilesInBackground(files).executeWith(Globals.TASK_EXECUTOR);
                 case CENTER -> {
-                    BibEntry entry = target.getEntry();
-                    switch (event.getTransferMode()) {
-                        case LINK -> {
-                            LOGGER.debug("Mode LINK"); // shift on win or no modifier
-                            importHandler.getLinker().addFilesToEntry(entry, files);
-                        }
-                        case MOVE -> {
-                            LOGGER.debug("Mode MOVE"); // alt on win
-                            importHandler.getLinker().moveFilesToFileDirAndAddToEntry(entry, files);
-                        }
-                        case COPY -> {
-                            LOGGER.debug("Mode Copy"); // ctrl on win
-                            importHandler.getLinker().copyFilesToFileDirAndAddToEntry(entry, files);
+                    try {
+                        BibEntry entry = target.getEntry();
+                        switch (event.getTransferMode()) {
+                            case LINK -> {
+                                LOGGER.debug("Mode LINK"); // shift on win or no modifier
+                                importHandler.getLinker().addFilesToEntry(entry, files);
+                            }
+                            case MOVE -> {
+                                LOGGER.debug("Mode MOVE"); // alt on win
+                                libraryTab.getIndexingTaskManager().setBlockingNewTasks(true);
+                                importHandler.getLinker().moveFilesToFileDirAndAddToEntry(entry, files);
+                                libraryTab.getIndexingTaskManager().setBlockingNewTasks(false);
```

Comment:
Instead of this `setBlockingNewTasks(true/false)` pattern, I would suggest to use the try resource solution: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html, i.e.
```
try (libraryTab.getIndexingTaskManager().blockingNewTasks()) {
   importHandler.getLinker().moveFilesToFileDirAndAddToEntry(entry, files);
}
```
In this way, you make sure that the `setBlockingNewTasks(false)` is always executed, even in case there is a (runtime) exception.

Reply:
So if I understand that right I would need to make the blocking() method of IndexingTaskManager return a 'closable', right? I could define that as a lambda that resets the flag.
Or is there a better way you are hinting at?

Reply:
That sounds good!

Reply:
Tried it and it was easy enough - so I think I got what you meant. Thanks for the hint!